lec02-basic-r.Rmd

```diff
@@ -1,4 +1,765 @@
+RStudio... You will be prompted to install a few packages, click Yes/OK. I have
+asked the IT services to make sure these packages are always installed on the
+computers so that we don't have to install them every time, hopefully they will
+do so soon. If you prefer, you can also bring your own laptop to class, which
+has the advantage that you can save things permanently and that you have
+everything setup, this is up to you. Let's go over the lesson objectives while
+we wait for these packages to install.
+
+
+## Learning Objectives
+
+* Define the following terms as they relate to R: object, assign, call,
+  function, arguments, options.
+* Create objects and assign values to them.
+* Use comments to inform script.
+* Do simple arithmetic operations in R using values and objects.
+* Call functions and use arguments to change their default options.
+* Inspect the content of vectors and manipulate their content.
+* Subset and extract values from vectors.
+* Correctly define and handle missing values in vectors.
+
+
+## Setting up the R Notebook
+
+Let's remove the template RStudio gives us, and add a title of our own.
+
+```yaml
+---
+title: Introduction to R
+---
+```
+
+This header block is called the YAML header. This is where we specify whether we
+want to convert this file to a HTML or PDF file. This will be discussed in more
+detail in another class. For now, we just care about including the lecture title
+here. For now, let's type out a note:
+
+> This lecture covers fundamental R usage, such as assigning values to a
+> variable, using functions, commenting code, and more.
+
+Under this sentence, we will insert our first code chunk. You insert a code
+chunk by either clicking the "Insert" button or pressing <kbd>Ctrl</kbd> +
+<kbd>Alt</kbd> + <kbd>i</kbd>. You could also type out the surrounding
+backticks, but this would take longer. To run a code chunk, you press the
+green arrow, or <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>Enter</kbd>.
+
+
+## Creating objects in R
+
+As we saw in our first class, you can get output from R simply by typing math in
+the console:
+
+```{r}
+3 + 5
+12 / 7
+```
+
+However, to do useful and interesting things, we need to assign _values_ to
+_objects_. Objects are also referred to as _variables_, and although there
+technically are minor differences between them[^objects_variables], we will use them interchangeably
+in this class. To create an object, we need to give it a name followed by the
+assignment operator `<-`, and the value we want to assign:
+
+[^objects_variables]https://cran.r-project.org/doc/manuals/r-release/R-lang.html#Objects).
+
+```{r}
+x <- 3
+x + 5
+```
+
+`<-` is the assignment operator. It assigns values on the right to objects on
+the left. So, after executing `x <- 3`, the value of `x` is `3`. The arrow can
+be read as 3 **goes into** `x`.[^equal_vs_arrow] In RStudio, typing
+<kbd>Alt</kbd> + <kbd>-</kbd> (push <kbd>Alt</kbd> at the same time as the
```

Comment:
The "at the same time" explanation should have gone above where the keyboard shortcuts were first shown.